bigbone/src/main/kotlin/social/bigbone/api/Scope.kt

```diff
@@ -3,22 +3,257 @@ package social.bigbone.api
+
+        /**
+         * Child scope of [WRITE].
+         */
+        WRITE_REPORTS("write:reports"),
+
+        /**
+         * Child scope of [WRITE].
+         */
+        WRITE_STATUSES("write:statuses"),
+
+        /**
+         * Grants access to Web Push API subscriptions.
+         */
+        PUSH("push"),
+
+        /**
+         * Used for moderation API. Requesting this scope will also grant its child scopes.
+         */
+        ADMIN_READ("admin:read"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_ACCOUNTS("admin:read:accounts"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_REPORTS("admin:read:reports"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_DOMAIN_ALLOWS("admin:read:domain_allows"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_DOMAIN_BLOCKS("admin:read:domain_blocks"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_IP_BLOCKS("admin:read:ip_blocks"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_EMAIL_DOMAIN_BLOCKS("admin:read:email_domain_blocks"),
+
+        /**
+         * Child scope of [ADMIN_READ].
+         */
+        ADMIN_READ_CANONICAL_EMAIL_BLOCKS("admin:read:canonical_email_blocks"),
+
+        /**
+         * Used for moderation API. Requesting this scope will also grant its child scopes.
+         */
+        ADMIN_WRITE("admin:write"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_ACCOUNTS("admin:write:accounts"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_REPORTS("admin:write:reports"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_DOMAIN_ALLOWS("admin:write:domain_allows"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_DOMAIN_BLOCKS("admin:write:domain_blocks"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_IP_BLOCKS("admin:write:ip_blocks"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_EMAIL_DOMAIN_BLOCKS("admin:write:email_domain_blocks"),
+
+        /**
+         * Child scope of [ADMIN_WRITE].
+         */
+        ADMIN_WRITE_CANONICAL_EMAIL_BLOCKS("admin:write:canonical_email_blocks"),
+
+        /**
+         * Grants access to manage relationships. Requesting this scope will also grant the child scopes
+         * [READ_BLOCKS], [WRITE_BLOCKS], [READ_FOLLOWS], [WRITE_FOLLOWS], [READ_MUTES] and [WRITE_MUTES].
+         */
+        @Deprecated(
+            message = "Deprecated since Mastodon 3.5.0 and will be removed in the future. Use child scopes of READ and WRITE instead.",
+            level = DeprecationLevel.WARNING
```

Comment:
As this is likely to be removed very soon, maybe we should go for `HIDDEN` instead so that new users don’t even see this at all and existing users get an error here?

```suggestion
            level = DeprecationLevel.HIDDEN
```

Reply:
My idea was to start properly deprecating things before completely breaking them later - but we've broken so many things that one more wouldn't make much of a difference. ;) If we're OK with this being even more of a breaking change, I could completely remove those two definitions, which I think would have the same effect as declaring them `HIDDEN` and thus unusable.

Reply:
Now that the whole enum definition has changed, users won't immediately see this deprecation warning anyway. Should the two definitions that are currently marked as deprecated be removed entirely?